paymentsheet/src/main/java/com/stripe/android/paymentsheet/forms/Form.kt

```diff
@@ -33,30 +36,38 @@ internal fun Form(
     formViewModel: FormViewModel,
 ) {
     val form = formViewModel.formDataObject
-    val focusRequesters = List(form.sections.size) { FocusRequester() }
+
+    // There is only a single field in a section, some of those might not require focus like
+    // country so we will create an extra one
+    var focusRequesterIndex = 0
+    val focusRequesters = List(formViewModel.getNumberTextFieldElements()) { FocusRequester() }
 
     Column(
         modifier = Modifier
             .fillMaxWidth(1f)
             .padding(16.dp)
     ) {
-        form.sections.forEachIndexed { index, section ->
+        form.sections.forEach { section ->
             val element = formViewModel.getElement(section.field)
             val error by element.errorMessage.asLiveData().observeAsState(null)
             val sectionErrorString =
                 error?.let { stringResource(it, stringResource(element.label)) }
 
             Section(sectionErrorString) {
-                TextField(
-                    label = element.label,
-                    textFieldElement = element,
-                    myFocus = focusRequesters[index],
-                    nextFocus = if (index == form.sections.size - 1) {
-                        null
-                    } else {
-                        focusRequesters[index + 1]
-                    },
-                )
+                if (element is TextFieldElement) {
```

Comment:
I would use `when` if you can make `Element` a `sealed interface`

Reply:
I will address it here: https://github.com/stripe/stripe-android/pull/3838#issuecomment-861049326